var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -28,6 +29,7 @@ class PyNumba(PythonPackage):
     # That's why it was chosen as an upper bound in the following depends_on
     # calls.  If newer versions maintain backwards compatibility, the calls
     # can be updated accordingly.
+    depends_on('py-llvmlite@0.34:0.35', type=('build', 'run'), when='@0.51.1')
```

Comment:
If you look at:
```python
min_llvmlite_version = "0.34.0.dev0"                                            
max_llvmlite_version = "0.35"        
...
    'llvmlite >={},<{}'.format(min_llvmlite_version, max_llvmlite_version),     
```
it actually has to be less than 0.35. So maybe this should be:
```suggestion
    depends_on('py-llvmlite@0.34.0:0.34.999', type=('build', 'run'), when='@0.51.1')
```
Can you check older versions too?